diff --git a/src/ghdl.cc b/src/ghdl.cc
index 8c8410c..71c3949 100644
--- a/src/ghdl.cc
+++ b/src/ghdl.cc
@@ -81,8 +81,6 @@ static std::string to_str_1(Sname name)
 		return to_str_1(get_sname_prefix(name)) + ':' + stringf("%u", get_sname_version(name));
 	case Sname_Field:
 		return to_str_1(get_sname_prefix(name)) + '[' + user_to_str(get_sname_suffix(name)) + ']';
-	case Sname_Index:
-		return to_str_1(get_sname_prefix(name)) + '[' + stringf("%d", get_sname_index(name)) + ']';
 	case Sname_User: {
 		Sname pfx = get_sname_prefix(name);
 		std::string s = user_to_str(get_sname_suffix(name));
